src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -1202,5 +1279,53 @@ public static void GenerateNupkgMetadataFile(string nupkgPath, string installPat
                     }
                 });
         }
+
+        private static T[] CancellableEnumaration<T>(IEnumerable<T> enumerable, ILogger log, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // .ToArray is necessary because previously it returns Array
+                // If enumaration happen several times then same I/O calls repeatedly called on same input, I/O calls are more expensive then memory.
+                return CancellableYieldEnumaration(enumerable, log, cancellationToken).ToArray();
+            }
+            catch (Exception e) when (e is not OperationCanceledException)
+            {
+                // On cancellation we bubble up exception to call stack.
+                // Otherwise return all or nothing. If no exception return all.
+                // Break on first exception with logging in order to keep previous experience, return empty array, but don't throw.
+                log.LogWarning(e.Message);
+            }
+
+            return Array.Empty<T>();
+        }
+
+        private static IEnumerable<T> CancellableYieldEnumaration<T>(IEnumerable<T> enumerable, ILogger log, CancellationToken cancellationToken)
```

Comment:
```suggestion
        private static IEnumerable<T> CancellableYieldEnumeration<T>(IEnumerable<T> enumerable, ILogger log, CancellationToken cancellationToken)
```

If you accept these, your build probably fails because you'd need to chat it in the places where you call these methods as well.